src/system_metrics/mongoose_system_metrics_collector.erl

```diff
@@ -106,66 +99,77 @@ get_cluster_size() ->
 
 get_version() ->
     case lists:keyfind(mongooseim, 1, application:which_applications()) of
-        false -> Version = none;
-        {_, _, Ver} -> Version = Ver
-    end,
-    #{report_name => cluster, key => mim_version, value => Version}.
+        {_, _, Version} ->
+            #{report_name => cluster, key => mim_version, value => Version};
+        _ ->
+            []
+    end.
 
 get_components() ->
     Domains = ejabberd_router:dirty_get_all_domains(),
-    Hosts = ejabberd_config:get_global_option(hosts),
-    Components = lists:flatten(
-                    lists:map(
-                        fun(Host) ->
-                            check_components(Host, Domains)
-                        end, Hosts)),
-    LenComponents = length(Components),
+    Components = [ejabberd_router:lookup_component(D, node()) || D <- Domains],
+    LenComponents = length(lists:flatten(Components)),
     #{report_name => cluster, key => number_of_components, value => LenComponents}.
 
-check_components(Host, Domains) ->
-    lists:flatten(
-        lists:map(
-            fun(Domain) ->
-                ejabberd_router:lookup_component(Domain, Host)
-            end, Domains)).
-
 get_api() ->
-    ModulesOptions = lists:flatten(get_service_option(ejabberd_cowboy, modules)),
-    ApiList = lists:usort(lists:map(
-                fun(Module)->
-                    element(3, Module)
-                end, ModulesOptions)),
+    ServiceOptions = get_service_option(ejabberd_cowboy),
+    ModulesOptions = lists:flatten([ Mod || {modules, Mod} <- ServiceOptions]),
```

Comment:
I have just realized that on this list there can be some custom modules, not provided by us. Similar situation as with `mod_*` modules. I think we need to address that, not necessarily in this PR.